dev_tools/modules.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2021 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import dataclasses
+import os
+import sys
+from typing import List, Dict, Any
+
+_FOLDER = 'folder'
+_PACKAGE_PATH = 'package-path'
+
+
+@dataclasses.dataclass
+class Module:
+    root: str
+    raw_setup: Dict[str, Any]
+
+    name: str = dataclasses.field(init=False)
+    version: str = dataclasses.field(init=False)
+    top_level_packages: List[str] = dataclasses.field(init=False)
+
+    def __post_init__(self):
```

Comment:
I'd suggest putting this in a classmethod instead, e.g.:
```python
@dataclasses.dataclass
class Module:
    root: str
    name: str
    version: str
    top_level_packages: List[str]

    @classmethod
    def from_setup_kwargs(**kwargs: Any) -> 'Module':
        ...
```
Then call that classmethod in `list_modules`.

Reply:
I'm not fully convinced...the `root` folder needs to be passed in as well, also, I like that the raw kwargs are stored....

Reply:
ack.